src/app/clients/client-stepper/client-general-step/client-general-step.component.html

```diff
@@ -1,11 +1,17 @@
 <form [formGroup]="createClientForm">
-
   <div fxLayout="row wrap" fxLayoutGap="2%" fxLayout.lt-md="column">
-
     <mat-form-field fxFlex="48%">
       <mat-label>Office</mat-label>
-      <mat-select required formControlName="officeId">
-        <mat-option *ngFor="let office of officeOptions" [value]="office.id">
+      <mat-select required formControlName="officeId" #office>
+        <mat-select-filter
+          *ngIf="office.focused"
+          [array]="officeOptions"
+          (filteredReturn)="officeOptionsSliced = $event"
+          [displayMember]="'name'"
+        >
+        </mat-select-filter>
+
+        <mat-option *ngFor="let office of officeOptions" [value]="office.id" [class.hide]="!isFiltered(office)">
```

Comment:
@gilclefes I was thinking what if we use [hidden] (to hide) or *ngIf (to remove) directive instead of [class.hide] and then remove the CSS styles, wdyt?

Reply:
> @gilclefes I was thinking what if we use [hidden] (to hide) or *ngIf (to remove) directive instead of [class.hide] and then remove the CSS styles, wdyt?

will check the [hidden] and confirm if it works fine, for the ngIf we cannot use it together with ngFor on the same element.

Reply:
[hidden] does not work for this after trying, not able to display the items based on the search parameter so we will need to use this [class.hide] for now.